src/Magnum/Math/Functions.h

```diff
@@ -34,6 +34,7 @@
 #include <cstdlib> /* std::div() */
 #include <type_traits>
 #include <utility>
+#include <limits>
```

Comment:
This adds about 1k preprocessed lines which isn't too much but also isn't too little. With the function not being templated anymore, you could use `~UnsignedInt{}` instead for the second assert.